config/src/main/java/com/alibaba/nacos/config/server/service/DynamicDataSourceConnectionPool.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.nacos.config.server.service;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.apache.commons.dbcp.BasicDataSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+import javax.sql.DataSource;
+import java.util.concurrent.TimeUnit;
+
+import static com.alibaba.nacos.config.server.service.BasicDataSourceServiceImpl.defaultIfNull;
+
+/**
+ * @author codewaltz1994
+ */
+
+@Component
+public class DynamicDataSourceConnectionPool {
+
+    private static final String DEFAULT_DATABASE_CONNECTION_POOL="dhcp";
+    @Autowired
+    Environment env;
+
+
+    public DataSource getDataSource(String jdbcDriverName, String jdbcUrl, String userName, String password){
+
+        String val = env.getProperty("db.datasourceCP");
```

Comment:
I think maybe you can refer to `org.springframework.boot.jdbc.DataSourceBuilder`, or use this class directly to build the data source. such as:
``` java
public T build() {
		Class<? extends DataSource> type = getType();
		DataSource result = BeanUtils.instantiateClass(type);
		maybeGetDriverClassName();
		bind(result);
		return (T) result;
	}
```
And for future extensions, you can consider extracting the custom properties of the connection pool, you can also refer to `org.springframework.boot.autoconfigure.jdbc.DataSourceConfiguration `, because no one knows how many data source connection pools there are in the future.
``` java

		@Bean
		@ConfigurationProperties(prefix = "db.pool")
		public NacosConnectionPoolMetadata nacosConnectionPoolMetadata() {
			
			return new NacosConnectionPoolMetadata();
		}

```
``` java
public class NacosConnectionPoolMetadata {

 private  Map<String,String> custom = new HashMap();

//setter getter
```

``` properties
db.num=2
db.url.0=jdbc:mysql://11.162.196.16:3306/nacos_devtest?characterEncoding=utf8&connectTimeout=1000&socketTimeout=3000&autoReconnect=true
db.url.1=jdbc:mysql://11.163.152.9:3306/nacos_devtest?characterEncoding=utf8&connectTimeout=1000&socketTimeout=3000&autoReconnect=true
db.user=nacos_devtest
db.password=nacos
db.type=org.apache.commons.dbcp.BasicDataSource
db.pool.initialSize=10
db.pool.maxActive=20

```

Reply:
OK, thanks for your advice. I will change this and  learn more about what you said. Thanks a lot.